src/lib.rs

```diff
@@ -129,6 +128,92 @@ mod errors;
 pub mod libtest;
 mod messages;
 
+macro_rules! str_newtype {
+    ($name:ident) => {
+        /// String newtype
+        ///
+        /// Based on [cargo-util-schema's string newtype] but with two crucial differences:
+        ///
+        /// - This newtype does not verify the wrapped string.
+        /// - This newtype allows comparison with arbitrary types that implement `AsRef<str>`.
+        ///
+        /// [cargo-util-schema's string newtype]: https://github.com/epage/cargo/blob/d8975d2901e132c02b3f6b1d107f2f50b275a058/crates/cargo-util-schemas/src/manifest/mod.rs#L1355-L1413
```

Comment:
This documentation is too general imo. I think we should add the docs to the macro invocation and tune it for the specific types so we can link to Cargo.toml documentation

Reply:
Yea, that's good. Tho i was hoping https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html to have some docs on these, I couldn't find any

Reply:
Ah fun. It'll make its way to stable then. I guess just mark the test as ignore with a comment linking to this PR